tests/integration/Traits/NotifableEagerLoadingTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+class NotifableEagerLoadingTest extends NotifynderTestCase
+{
+    public function testGetNotificationRelationReadsConfigurationParameterIfNothingIsPassed()
+    {
+        $user = $this->createUser();
+        $this->sendNotificationTo($user);
+
+        $notification = $user->getNotificationRelation()->first();
+        $this->assertModelHasNoLoadedRelations($notification, ['category', 'from', 'to']);
+    }
+
+    public function testGetNotificationRelationCanEagerLoadAllRelationsIfTrueIsPassed()
+    {
+        $user = $this->createUser();
+        $this->sendNotificationTo($user);
+
+        $notification = $user->getNotificationRelation(true)->first();
+
+        $this->assertModelHasLoadedRelations($notification, ['category', 'from', 'to']);
+    }
+
+    public function testGetNotificationRelationCanEagerLoadASubsetOfRelationsIfAnArrayIsPassed()
+    {
+        $user = $this->createUser();
+        $this->sendNotificationTo($user);
+
+        $notification = $user->getNotificationRelation(['category'])->first();
+
+        $this->assertModelHasLoadedRelations($notification, ['category']);
+        $this->assertModelHasNoLoadedRelations($notification, ['from', 'to']);
+    }
+
+    public function testGetNotificationRelationDoesnotEagerLoadIfConfigurationParameterIsMissing()
+    {
+        $user = $this->createUser();
+        $this->sendNotificationTo($user);
+
+        $config = app('notifynder.config');
+        $config->forget('eager_load');
+
+        $notification = $user->getNotificationRelation()->first();
+        $this->assertModelHasNoLoadedRelations($notification, ['category', 'from', 'to']);
+    }
+
+    public function testGetNotificationsReadsEagerLoadConfigurationParameter()
+    {
+        $user = $this->createUser();
+        $this->sendNotificationTo($user);
+
+        $config = app('notifynder.config');
+
+        $config->set('eager_load', true);
+        $notifications = $user->getNotifications();
+        $this->assertModelHasLoadedRelations($notifications[0], ['category', 'from', 'to']);
+
+        $config->set('eager_load', false);
+        $notifications = $user->getNotifications();
+        $this->assertModelHasNoLoadedRelations($notifications[0], ['category', 'from', 'to']);
+
+        $config->set('eager_load', ['category', 'from']);
+        $notifications = $user->getNotifications();
+        $this->assertModelHasLoadedRelations($notifications[0], ['category', 'from']);
+        $this->assertModelHasNoLoadedRelations($notifications[0], ['to']);
+    }
+
+    private function assertModelHasLoadedRelations($model, $relationNames = [])
```

Comment:
Although it is just a helper test function, I will modify it to be protected for the the sake of Open/Closed Principle